app/controllers/spree/product_questions_controller.rb

```diff
@@ -1,24 +1,26 @@
-class Spree::ProductQuestionsController < Spree::Api::V2::ResourceController
-  before_action :load_data
+if Spree::Core::Engine.frontend_available?
```

Comment:
we will run into issue while running into problem as it is expected that class name is defined according to name of controller..

Reply:
```
module Spree
  if Spree::Core::Engine.frontend_available?
    class ProductQuestionsController < StoreController
      before_action :load_data

      def create
        question = @product.product_questions.new(allowed_params)
        question.user_id = current_spree_user.id
        if question.save
          flash[:notice] = t('question.sent')
          redirect_to @product
        else
          flash[:error] = question.errors.full_messages.to_sentence
          redirect_to @product
        end
      end

      private

      def load_data
        @product = Spree::Product.find(allowed_params[:product_id])
      end

      def allowed_params
        params.require(:product_question).permit(:content, :product_id)
      end
    end
  else
    class ProductQuestionsController
      # Dummy class to satisfy Zeitwerk autoloading
    end
  end
end
```
@mausamp I think this kind of approach would be better so we dont violate zeitwerk autoloading as well..
let me know your thoughts, here we are defining dummy ProductQuestionsController when frontend is not available in the project